src/uct/ib/ud/base/ud_iface.c

```diff
@@ -310,12 +311,19 @@ ucs_status_t uct_ud_iface_complete_init(uct_ud_iface_t *iface)
         goto err;
     }
 
-    status = uct_ib_iface_event_fd_get(&iface->super.super.super, &event_fd);
-    if (status != UCS_OK) {
-        goto err_twheel_cleanup;
-    }
-
     if (iface->async.event_cb != NULL) {
+        if (!dev->has_cq_notify) {
+            ucs_warn("Failed to set callback for async event handling. Device %s "
```

Comment:
1. smallcase
2. make it shorter "%s: async event handling is not supported". and print it during device init
3. probably should be ucs_debug